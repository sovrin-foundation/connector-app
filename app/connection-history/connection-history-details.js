// @flow
import React, { PureComponent } from 'react'
import { StyleSheet, Dimensions, Platform, ScrollView } from 'react-native'
import { createStackNavigator } from 'react-navigation'
import type { ConnectionHistoryDetailsProps } from './type-connection-history'
import {
  Container,
  CustomView,
  CustomText,
  CustomList,
  headerStyles,
  ConnectionTheme,
  CustomDate,
  Icon,
  CustomSafeAreaView,
} from '../components'
import {
  connectionHistoryDetailsRoute,
  connectionHistoryRoute,
} from '../common'
import {
  OFFSET_1X,
  OFFSET_3X,
  OFFSET_9X,
  OFFSET_2X,
  iPhoneXHeight,
} from '../common/styles/constant'

const styles = StyleSheet.create({
  headerContent: {
    marginTop: OFFSET_1X / 2,
  },
  headerDate: {
    marginBottom: OFFSET_3X / 2,
  },
  headerContainer: {
    minHeight: 109,
  },
})

export class ConnectionHistoryDetails extends PureComponent<
  ConnectionHistoryDetailsProps,
  void
> {
  static navigationOptions = ({
    navigation: {
      goBack,
      state: {
        params: {
          action,
          name,
          timestamp,
          theme,
          senderName,
          image,
          senderDID,
        },
      },
    },
  }) => ({
    headerLeft: (
      <CustomView>
        <Icon
          testID={'history-details-back-arrow'}
          iconStyle={[headerStyles.headerLeft]}
          small
          src={require('../images/icon_backArrow_white.png')}
          resizeMode="contain"
          onPress={() => goBack(null)}
        />
      </CustomView>
    ),
    headerTitle: (
      <Container clearBg center style={[headerStyles.centerTitle]}>
        {action && (
          <CustomText transparentBg center>
            {action}
          </CustomText>
        )}
        {name && (
          <CustomText
            style={[styles.headerContent]}
            transparentBg
            semiBold
            center
            numberOfLines={1}
          >
            {name}
          </CustomText>
        )}

        {timestamp && (
          <CustomDate
            transparentBg
            h7
            center
            uppercase
            style={[styles.headerContent, styles.headerDate]}
          >
            {timestamp}
          </CustomDate>
        )}
      </Container>
    ),
    headerRight: <Container />,
    headerStyle: [
      headerStyles.header,
      styles.headerContainer,
      { backgroundColor: theme },
    ],
  })

  getContainerStyle = (deviceClass: string) => {
    if (deviceClass === 'IphoneX') {
      return {
        marginTop: OFFSET_2X,
      }
    }
    return {
      marginTop: OFFSET_9X / 2,
    }
  }

  render() {
    const { height } = Dimensions.get('window')

    const deviceClass =
      Platform.OS === 'ios'
        ? height === iPhoneXHeight ? 'IphoneX' : 'ios'
        : 'android'

    if (this.props.navigation.state) {
      const { type, data, claimMap } = this.props.navigation.state.params
      const listType = type === 'CLAIM' ? 'center' : 'end'

      return (
        <Container style={[this.getContainerStyle(deviceClass)]}>
          {data && (
            <ScrollView>
              <CustomList items={data} type={listType} claimMap={claimMap} />
            </ScrollView>
          )}
        </Container>
      )
    }

    // TODO: Handle scenario if no navigation props are passed
    return <Container />
  }
}

export default createStackNavigator({
  [connectionHistoryDetailsRoute]: {
    screen: ConnectionHistoryDetails,
  },
})
