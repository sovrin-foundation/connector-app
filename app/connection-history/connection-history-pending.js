// @flow
import React, { PureComponent } from 'react'
import { connect } from 'react-redux'
import { Container } from '../components'
import type { ReactNavigation } from '../common/type-common'
import type { Store } from '../store/type-store'
import { ClaimRequestStatusModal } from '../claim-offer/claim-request-modal'
import { CLAIM_REQUEST_STATUS } from '../claim-offer/type-claim-offer'

type ConnectionHistoryPendingProps = {} & ReactNavigation

export class ConnectionHistoryPending extends PureComponent<
  ConnectionHistoryPendingProps,
  void
> {
  close = () => {
    this.props.navigation.goBack()
  }

  render() {
    const { payload } = this.props.navigation.state
      ? this.props.navigation.state.params
      : {}

    return (
      <Container fifth>
        <ClaimRequestStatusModal
          fromConnectionHistory
          claimRequestStatus={CLAIM_REQUEST_STATUS.SEND_CLAIM_REQUEST_SUCCESS}
          payload={payload}
          onContinue={this.close}
          senderLogoUrl={payload.senderLogoUrl}
          isPending={true}
          message1="As soon as"
          message3="signs and issues the credential"
          message5="to you, it will appear here."
        />
      </Container>
    )
  }
}
