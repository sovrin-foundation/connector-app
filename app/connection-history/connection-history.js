// @flow
import React, { Component } from 'react'
import {
  Alert,
  ScrollView,
  StyleSheet,
  Dimensions,
  View,
  StatusBar,
} from 'react-native'
import { connect } from 'react-redux'
import { List, ListItem } from 'react-native-elements'
import moment from 'moment'
import groupBy from 'lodash.groupby'
import { bindActionCreators } from 'redux'
import { updateStatusBarTheme } from '../../app/store/connections-store'
import {
  InfoSectionList,
  ClaimProofHeader,
  Container,
  ConnectionTheme,
  CustomView,
  Icon,
  CustomText,
  Separator,
  CustomDate,
  headerStyles,
} from '../components'
import { dimGray, OFFSET_9X, white } from '../../app/common/styles'
import {
  homeRoute,
  connectionHistoryRoute,
  connectionHistoryDetailsRoute,
  connectionHistoryPendingRoute,
} from '../common/route-constants'
import {
  color,
  OFFSET_1X,
  OFFSET_3X,
  alertCancel,
} from '../common/styles/constant'
import type { Store } from '../store/type-store'
import type {
  ConnectionHistoryItem,
  ConnectionHistoryProps,
  ConnectionHistoryState,
  ConnectionHistoryEvent,
} from './type-connection-history'
import {
  HISTORY_EVENT_STATUS,
  HISTORY_EVENT_OCCURRED,
} from './type-connection-history'
import type { ReactNavigation } from '../common/type-common'
import showDID from '../components/show-pairwise-info'
import {
  SEND_CLAIM_REQUEST,
  CLAIM_REQUEST_STATUS,
  SEND_CLAIM_REQUEST_SUCCESS,
} from '../claim-offer/type-claim-offer'
import { deleteConnectionAction } from '../store/connections-store'
import { getConnection, getConnectionTheme } from '../store/store-selector'
import { Dot as BadgeDot } from '../components/badges-dot'

import debounce from 'lodash.debounce'
import { getUnseenMessages } from '../store/store-selector'
import { goToUIScreen } from '../push-notification/push-notification-store'
import Color from 'color'
import { getStatusBarStyle } from '../components/custom-header/custom-header'

const statusMsg = {
  ['PENDING']: 'Pending',
  ['CONNECTED']: 'Established on',
  ['RECEIVED']: 'Accepted on',
  ['ACCEPTED & SAVED']: 'Accepted on',
  ['SHARED']: 'Sent on',
  ['PROOF RECEIVED']: 'New request to share',
  ['CLAIM OFFER RECEIVED']: 'New credential offer',
}

const historyIcons = {
  ['PENDING']: require('../images/received.png'),
  ['CONNECTED']: require('../images/linked.png'),
  ['RECEIVED']: require('../images/received.png'),
  ['CLAIM OFFER RECEIVED']: require('../images/received.png'),
  ['PROOF RECEIVED']: require('../images/received.png'),
  ['ACCEPTED & SAVED']: require('../images/received.png'),
  ['SHARED']: require('../images/sent.png'),
}

const historyShowUI = ['CLAIM OFFER RECEIVED', 'PROOF RECEIVED']

const HistoryTitle = ({ action, name, theme }) => (
  <CustomView>
    <CustomView>
      <CustomText
        h5
        semiBold
        uppercase
        bg="fifth"
        style={[styles.listItemAction, { color: theme }]}
      >
        {action && historyShowUI.indexOf(action) > -1
          ? statusMsg[action]
          : action}
      </CustomText>
    </CustomView>
    <CustomView>
      {name && (
        <CustomText
          h5
          semiBold
          bg="fifth"
          style={[styles.listItemTitleEvent, styles.listItemAction]}
        >
          {name}
        </CustomText>
      )}
    </CustomView>
  </CustomView>
)

const HistoryBody = ({ action, timestamp }) => {
  return (
    <CustomView row>
      <CustomDate h7 uppercase bg="fifth" style={[styles.listItemBody]}>
        {timestamp}
      </CustomDate>
    </CustomView>
  )
}

export class ConnectionHistory extends Component<
  ConnectionHistoryProps,
  ConnectionHistoryState
> {
  state = {
    disableTaps: false,
  }

  connectionDetailHandlerDebounce = debounce(
    ({ h, activeConnectionThemePrimary, senderName, image, senderDID }) => {
      this.connectionDetailHandler({
        h,
        activeConnectionThemePrimary,
        senderName,
        image,
        senderDID,
      })
    },
    300,
    { leading: false, trailing: true }
  )

  componentDidMount() {
    this.props.updateStatusBarTheme(this.props.activeConnectionThemePrimary)
  }

  closeDebounce = () => {
    if (!this.state.disableTaps) {
      const { navigation } = this.props

      navigation.goBack(null)
      this.setState({ disableTaps: true })
    }
  }

  connectionDetailHandler = (history: any) => {
    if (history.h.action === HISTORY_EVENT_STATUS[SEND_CLAIM_REQUEST_SUCCESS]) {
      this.props.navigation.navigate(connectionHistoryPendingRoute, {
        payload: history.h.originalPayload.payload,
      })
    } else {
      this.props.navigation.navigate(connectionHistoryDetailsRoute, {
        ...history.h,
        theme: history.activeConnectionThemePrimary,
        senderName: history.senderName,
        image: history.image,
        senderDID: history.senderDID,
        claimMap: this.props.claimMap,
      })
    }
  }

  getHistoryIcons = (action: string, showBadge?: boolean) => {
    return (
      <CustomView center>
        {}
        {showBadge && (
          <View style={[styles.badgeDotStyle]}>
            <BadgeDot size="small" />
          </View>
        )}
        <Icon
          src={historyIcons[action]}
          iconStyle={[{ tintColor: this.props.activeConnectionThemePrimary }]}
          small
        />
      </CustomView>
    )
  }

  handleRedirectionScreens(
    h: ConnectionHistoryEvent,
    activeConnectionThemePrimary: string,
    senderName: string,
    image: string,
    senderDID: string
  ) {
    if (h.showBadge || historyShowUI.indexOf(h.action) != -1) {
      this.props.goToUIScreen(
        h.originalPayload.type,
        h.originalPayload.payloadInfo.uid,
        this.props.navigation
      )
    } else {
      this.connectionDetailHandlerDebounce({
        h,
        activeConnectionThemePrimary,
        senderName,
        image,
        senderDID,
      })
    }
  }

  onDeleteConnection = (senderName: string, senderDID: string) => {
    Alert.alert(
      `Delete ${senderName}?`,
      `Are you sure you want to delete ${senderName} as a connection? You will still keep the information they have given you, but you will not be able to contact them without re-establishing a new connection. Proceed?`,
      [alertCancel, { text: 'Delete', onPress: () => this.delete(senderDID) }]
    )
  }

  delete = (senderDID: string) => {
    this.props.deleteConnectionAction(senderDID)
    this.closeDebounce()
  }

  render() {
    const { width } = Dimensions.get('window')
    if (this.props.navigation.state) {
      const {
        senderName,
        image,
        senderDID,
        identifier,
      } = this.props.navigation.state.params
      const { activeConnectionThemePrimary, connectionHistory } = this.props
      const testID = 'connection-history'
      const logoUri = image
        ? { uri: image }
        : require('../images/logo_sovrin.png')

      const historySenderDIDs = Object.keys(connectionHistory)
      const historyList = historySenderDIDs.map((sdid, i) => {
        const historyItems = connectionHistory[sdid].map((h, i) => {
          const itemProps = {
            avatar: this.getHistoryIcons(h.action, h.showBadge),
            key: h.id,
            title: <HistoryTitle {...h} theme={activeConnectionThemePrimary} />,
            subtitle: <HistoryBody {...h} />,
            chevronColor: color.bg.fifth.font.fifth,
            avatarStyle: styles.avatarStyle,
            avatarOverlayContainerStyle: styles.avatarOverlayContainerStyle,
            containerStyle: styles.listItemContainer,
            hideChevron: false,
            rightIcon:
              h.action === HISTORY_EVENT_STATUS[SEND_CLAIM_REQUEST_SUCCESS] ? (
                <Icon src={require('../images/e15c.1.png')} small />
              ) : (
                {
                  name: 'chevron-right',
                  color: dimGray,
                }
              ),
            onPress: () => {
              this.handleRedirectionScreens(
                h,
                activeConnectionThemePrimary,
                senderName,
                image,
                senderDID
              )
            },
          }

          if (h.action === 'CONNECTED') {
            itemProps.hideChevron = true
            delete itemProps.onPress
          }

          return <ListItem {...itemProps} />
        })
        const history = (
          <CustomView key={i}>
            <CustomText
              h4
              semiBold
              bg="septenary"
              style={[styles.listItemLabel]}
            >
              {sdid}
            </CustomText>
            {historyItems}
          </CustomView>
        )
        return history
      })

      return (
        <Container fifth>
          <StatusBar
            backgroundColor={this.props.activeConnectionThemePrimary}
            barStyle={getStatusBarStyle(
              this.props.activeConnectionThemePrimary
            )}
          />
          <CustomView
            style={[
              { backgroundColor: this.props.activeConnectionThemePrimary },
            ]}
          >
            <ClaimProofHeader
              message={senderName}
              onClose={this.closeDebounce}
              logoUrl={image}
              testID={testID}
              containerStyle={{ backgroundColor: 'transparent' }}
              textContainerStyle={[
                styles.textContainerStyle,
                { backgroundColor: 'transparent' },
              ]}
              messageStyle={[
                styles.senderName,
                { backgroundColor: 'transparent' },
              ]}
            >
              <CustomView
                fifth
                hCenter
                style={[
                  headerStyles.headerLogoContainer,
                  { backgroundColor: 'transparent' },
                ]}
              >
                <Icon
                  absolute="TopRight"
                  src={require('../images/close_white.png')}
                  small
                  testID={`${testID}-icon-close`}
                  onPress={this.closeDebounce}
                  iconStyle={[styles.headerCloseIcon]}
                  style={[styles.headerIconContainer]}
                />
                <Icon
                  center
                  halo
                  extraLarge
                  resizeMode="cover"
                  src={logoUri}
                  style={[styles.issuerLogo]}
                  iconStyle={[styles.issuerLogoIcon]}
                  testID={`${testID}-issuer-logo`}
                  onLongPress={() => showDID(senderDID, identifier)}
                  backgroundRoundWhite
                />
                <Icon
                  absolute="TopLeft"
                  src={require('../images/delete.png')}
                  small
                  iconStyle={[styles.headerDeleteIcon]}
                  style={[styles.headerIconContainer]}
                  resizeMode="contain"
                  testID={`${testID}-icon-delete`}
                  onPress={() => {
                    this.onDeleteConnection(senderName, senderDID)
                  }}
                />
              </CustomView>
            </ClaimProofHeader>
          </CustomView>
          <ScrollView>
            <List containerStyle={styles.listContainer}>{historyList}</List>
          </ScrollView>
        </Container>
      )
    }
  }
}

const mapStateToProps = (state: Store, props: ReactNavigation) => {
  let connectionHistory: ConnectionHistoryEvent[] =
    state.history.data && props.navigation.state
      ? state.history.data[props.navigation.state.params.senderDID]
      : []

  if (connectionHistory) {
    connectionHistory = connectionHistory.filter(history => {
      if (statusMsg[history.action]) {
        return history
      }
    })
  }

  const unSeenMessages = getUnseenMessages(state)
  if (connectionHistory) {
    connectionHistory = connectionHistory.map(history => {
      if (
        history.originalPayload &&
        history.originalPayload.payloadInfo &&
        history.originalPayload.payloadInfo.uid &&
        unSeenMessages[history.remoteDid] &&
        unSeenMessages[history.remoteDid].length > 0 &&
        unSeenMessages[history.remoteDid].indexOf(
          history.originalPayload.payloadInfo.uid
        ) >= 0
      ) {
        history.showBadge = true
      } else {
        history.showBadge = false
      }
      return history
    })
  }

  const themeForLogo = getConnectionTheme(
    state,
    props.navigation.state ? props.navigation.state.params.image : ''
  )

  return {
    activeConnectionThemePrimary: themeForLogo.primary,
    activeConnectionThemeSecondary: themeForLogo.secondary,
    connectionHistory: groupBy(
      connectionHistory.sort((a, b) => {
        return new Date(b.timestamp) - new Date(a.timestamp)
      }),
      history => moment(history.timestamp).format('MMMM YYYY')
    ),
    claimMap: state.claim.claimMap,
  }
}

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    { deleteConnectionAction, updateStatusBarTheme, goToUIScreen },
    dispatch
  )

export default connect(mapStateToProps, mapDispatchToProps)(ConnectionHistory)

const styles = StyleSheet.create({
  headerCloseIcon: {
    marginRight: 15,
  },
  headerIconContainer: {
    zIndex: 2,
  },
  headerDeleteIcon: {
    marginLeft: 15,
  },
  issuerLogo: {
    ...StyleSheet.absoluteFillObject,
    alignSelf: 'center',
    zIndex: 1,
  },
  textContainerStyle: {
    paddingBottom: 0,
    marginBottom: OFFSET_3X / 2,
  },
  senderName: {
    marginTop: OFFSET_1X / 2,
    color: color.bg.primary.font.primary,
  },
  issuerLogoIcon: {
    borderRadius: 40,
  },
  listContainer: {
    marginTop: 0,
    borderBottomWidth: 0,
  },
  listItemLabel: {
    paddingLeft: OFFSET_3X / 2,
    paddingVertical: OFFSET_1X,
  },
  listItemContainer: {
    borderBottomWidth: 1,
  },
  listItemAction: {
    marginLeft: OFFSET_1X,
    marginBottom: OFFSET_1X / 2,
  },
  listItemTitleEvent: {
    color: color.bg.fifth.font.fifth,
  },
  listItemBody: {
    marginLeft: OFFSET_1X,
  },
  avatarStyle: {
    width: 17,
    height: 17,
    marginVertical: OFFSET_1X,
    resizeMode: 'contain',
  },
  avatarOverlayContainerStyle: { backgroundColor: 'transparent' },
  badgeDotStyle: {
    marginBottom: 5,
  },
  issuerLogoHaloStyle: {
    backgroundColor: white,
  },
})
