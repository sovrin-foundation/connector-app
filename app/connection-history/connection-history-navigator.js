// @flow
import { createStackNavigator } from 'react-navigation'

import {
  connectionHistoryDetailsRoute,
  connectionHistoryPendingRoute,
  connectionHistoryRoute,
} from '../common/route-constants'
import ConnectionHistory from './connection-history'
import ConnectionHistoryDetails from './connection-history-details'
import { ConnectionHistoryPending } from './connection-history-pending'

export default createStackNavigator(
  {
    [connectionHistoryRoute]: {
      screen: ConnectionHistory,
    },
    [connectionHistoryDetailsRoute]: {
      screen: ConnectionHistoryDetails,
    },
    [connectionHistoryPendingRoute]: {
      screen: ConnectionHistoryPending,
    },
  },
  {
    headerMode: 'none',
  }
)
