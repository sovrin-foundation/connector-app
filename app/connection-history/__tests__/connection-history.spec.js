// @flow
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { ConnectionHistory } from '../connection-history'
import { color } from '../../common/styles'
import {
  getStore,
  getNavigation,
  connection,
  connectionHistory,
  activeConnectionThemePrimary,
  activeConnectionThemeSecondary,
} from '../../../__mocks__/static-data'

describe('<ConnectionHistory />', () => {
  const store = getStore()

  function props() {
    return {
      navigation: getNavigation({
        image: 'https://test.com/image.png',
        senderName: 'Evernym',
      }),
      activeConnectionThemePrimary,
      activeConnectionThemeSecondary,
      connectionHistory,
      connection,
      claimMap: undefined,
      updateStatusBarTheme: () => {},
      deleteConnectionAction: jest.fn(),
      goToUIScreen: jest.fn(),
    }
  }

  it('should ConnectionHistory render properly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <ConnectionHistory {...props()} />
      </Provider>
    )
    expect(component).toMatchSnapshot()
  })
})
