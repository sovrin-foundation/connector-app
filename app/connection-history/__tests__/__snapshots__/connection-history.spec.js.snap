// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConnectionHistory /> should ConnectionHistory render properly 1`] = `
<View
  fifth={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(119, 119, 119, 1.0)",
      }
    }
  >
    <RCTSafeAreaView
      style={Object {}}
    >
      <View
        fifth={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "marginBottom": -10,
            "paddingTop": 30,
            "shadowColor": "#777777",
            "shadowOffset": Object {
              "height": 20,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 10,
            "zIndex": 4,
          }
        }
      >
        <View
          fifth={true}
          hCenter={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "height": 90,
              "justifyContent": "center",
            }
          }
        >
          <View
            absolute="TopRight"
            accessibilityComponentType={undefined}
            accessibilityLabel="connection-history-icon-close-touchable"
            accessibilityTraits={undefined}
            accessible={false}
            clearBg={true}
            hitSlop={
              Object {
                "bottom": 30,
                "left": 30,
                "right": 30,
                "top": 30,
              }
            }
            iconStyle={
              Array [
                Object {
                  "marginRight": 15,
                },
              ]
            }
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            small={true}
            src={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 2,
              }
            }
            testID="connection-history-icon-close-touchable"
          >
            <View
              style={Object {}}
            >
              <Image
                accessibilityLabel="connection-history-icon-close"
                accessible={true}
                resizeMode="contain"
                source={1}
                style={
                  Array [
                    Object {
                      "height": 20,
                      "width": 20,
                    },
                    null,
                    Object {
                      "marginRight": 15,
                    },
                  ]
                }
                testID="connection-history-icon-close"
              />
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel="connection-history-issuer-logo-touchable"
            accessibilityTraits={undefined}
            accessible={false}
            backgroundRoundWhite={true}
            center={true}
            clearBg={true}
            extraLarge={true}
            halo={true}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            iconStyle={
              Array [
                Object {
                  "borderRadius": 40,
                },
              ]
            }
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            resizeMode="cover"
            src={
              Object {
                "uri": "https://test.com/image.png",
              }
            }
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 1,
              }
            }
            testID="connection-history-issuer-logo-touchable"
          >
            <View
              center={true}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(0,0,0,0.07)",
                  "borderRadius": 50,
                  "height": 100,
                  "justifyContent": "center",
                  "width": 100,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#ffffff",
                    },
                    Object {
                      "borderRadius": 40,
                    },
                  ]
                }
              >
                <Image
                  accessibilityLabel="connection-history-issuer-logo"
                  accessible={true}
                  resizeMode="cover"
                  source={
                    Object {
                      "uri": "https://test.com/image.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "height": 80,
                        "width": 80,
                      },
                      Object {
                        "borderRadius": 40,
                      },
                      Object {
                        "borderRadius": 40,
                      },
                    ]
                  }
                  testID="connection-history-issuer-logo"
                />
              </View>
            </View>
          </View>
          <View
            absolute="TopLeft"
            accessibilityComponentType={undefined}
            accessibilityLabel="connection-history-icon-delete-touchable"
            accessibilityTraits={undefined}
            accessible={false}
            clearBg={true}
            hitSlop={
              Object {
                "bottom": 30,
                "left": 30,
                "right": 30,
                "top": 30,
              }
            }
            iconStyle={
              Array [
                Object {
                  "marginLeft": 15,
                },
              ]
            }
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            resizeMode="contain"
            small={true}
            src={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "left": 0,
                "position": "absolute",
                "top": 0,
                "zIndex": 2,
              }
            }
            testID="connection-history-icon-delete-touchable"
          >
            <View
              style={Object {}}
            >
              <Image
                accessibilityLabel="connection-history-icon-delete"
                accessible={true}
                resizeMode="contain"
                source={1}
                style={
                  Array [
                    Object {
                      "height": 20,
                      "width": 20,
                    },
                    null,
                    Object {
                      "marginLeft": 15,
                    },
                  ]
                }
                testID="connection-history-icon-delete"
              />
            </View>
          </View>
        </View>
        <View
          center={true}
          fifth={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
              "marginBottom": 15,
              "marginHorizontal": 10,
              "marginTop": 5,
              "paddingBottom": 0,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "fontFamily": "Lato",
                },
                Object {
                  "backgroundColor": "#ffffff",
                  "color": "#777777",
                },
                Object {
                  "fontWeight": "600",
                },
                Object {
                  "textAlign": "center",
                },
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                Array [
                  Object {
                    "color": "#ffffff",
                    "marginTop": 5,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ],
              ]
            }
          >
            Evernym
          </Text>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderColor": "#bbb",
              "borderTopWidth": 1,
              "marginTop": 20,
            },
            Object {
              "borderBottomWidth": 0,
              "marginTop": 0,
            },
          ]
        }
      >
        <View
          style={Object {}}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 20,
                },
                Object {
                  "fontFamily": "Lato",
                },
                Object {
                  "backgroundColor": "#f0f0f0",
                  "color": "#333333",
                },
                Object {
                  "fontWeight": "600",
                },
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                Object {
                  "paddingLeft": 15,
                  "paddingVertical": 10,
                },
              ]
            }
          >
            September 2017
          </Text>
          <View
            avatar={
              <CustomView
                center={true}
              >
                <Icon
                  iconStyle={
                    Array [
                      Object {
                        "tintColor": "rgba(119, 119, 119, 1.0)",
                      },
                    ]
                  }
                  small={true}
                  src={1}
                />
              </CustomView>
            }
            disabled={false}
            onLongPress={undefined}
            onPress={undefined}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderBottomColor": "#bbb",
                  "borderBottomWidth": 1,
                  "paddingBottom": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                },
                Object {
                  "borderBottomWidth": 1,
                },
                false,
                false,
              ]
            }
            underlayColor="white"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginLeft": 10,
                  },
                  undefined,
                ]
              }
            >
              <View
                center={true}
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  clearBg={true}
                  iconStyle={
                    Array [
                      Object {
                        "tintColor": "rgba(119, 119, 119, 1.0)",
                      },
                    ]
                  }
                  small={true}
                  src={1}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                    }
                  }
                >
                  <View
                    style={Object {}}
                  >
                    <Image
                      resizeMode="contain"
                      source={1}
                      style={
                        Array [
                          Object {
                            "height": 20,
                            "width": 20,
                          },
                          null,
                          Object {
                            "tintColor": "rgba(119, 119, 119, 1.0)",
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={undefined}
                >
                  <View>
                    <View
                      style={Object {}}
                    >
                      <View
                        style={Object {}}
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "fontSize": 17,
                              },
                              Object {
                                "fontFamily": "Lato",
                              },
                              Object {
                                "backgroundColor": "#ffffff",
                                "color": "#777777",
                              },
                              Object {
                                "fontWeight": "600",
                              },
                              null,
                              null,
                              null,
                              null,
                              null,
                              null,
                              null,
                              null,
                              null,
                              null,
                              Object {
                                "marginBottom": 5,
                                "marginLeft": 10,
                              },
                              Object {
                                "color": "rgba(119, 119, 119, 1.0)",
                              },
                            ]
                          }
                        >
                          CONNECTED
                        </Text>
                      </View>
                      <View
                        style={Object {}}
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "fontSize": 17,
                              },
                              Object {
                                "fontFamily": "Lato",
                              },
                              Object {
                                "backgroundColor": "#ffffff",
                                "color": "#777777",
                              },
                              Object {
                                "fontWeight": "600",
                              },
                              null,
                              null,
                              null,
                              null,
                              null,
                              null,
                              null,
                              null,
                              null,
                              null,
                              Object {
                                "color": "#333333",
                              },
                              Object {
                                "marginBottom": 5,
                                "marginLeft": 10,
                              },
                            ]
                          }
                        >
                          Enterprise name
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={undefined}
                >
                  <View>
                    <View
                      row={true}
                      style={
                        Object {
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={false}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "fontSize": 12,
                            },
                            Object {
                              "fontFamily": "Lato",
                            },
                            Object {
                              "backgroundColor": "#ffffff",
                              "color": "#777777",
                            },
                            null,
                            null,
                            null,
                            null,
                            null,
                            null,
                            null,
                            null,
                            null,
                            null,
                            null,
                            Object {
                              "marginLeft": 10,
                            },
                          ]
                        }
                      >
                        2018-01-10T04:32:43+05:30
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
